examples/assets/scripts/parsers/hdr-texture.js

```diff
@@ -0,0 +1,222 @@
+function Reader(data) {
+    this.view = new DataView(data);
+    this.offset = 0;
+}
+
+Object.assign(Reader.prototype, {
+    readLine: function () {
+        var view = this.view;
+        var result = "";
+        while (true) {
+            if (this.offset >= view.byteLength) {
+                break;
+            }
+
+            var c = String.fromCharCode(this.readUint8());
+            if (c === '\n') {
+                break;
+            }
+            result += c;
+        }
+        return result;
+    },
+
+    readUint8: function () {
+        return this.view.getUint8(this.offset++);
+    },
+
+    readUint8s: function (result) {
+        var view = this.view;
+        for (var i = 0; i < result.length; ++i) {
+            result[i] = view.getUint8(this.offset++);
+        }
+    },
+
+    peekUint8s: function (result) {
+        var view = this.view;
+        for (var i = 0; i < result.length; ++i) {
+            result[i] = view.getUint8(this.offset + i);
+        }
+    }
+});
+
+/**
+ * @class
+ * @name pc.HdrParser
+ * @implements {pc.TextureParser}
+ * @classdesc Texture parser for hdr files.
+ */
+var HdrParser = function (registry, retryRequests) {
+    this.retryRequests = retryRequests;
+};
+
+Object.assign(HdrParser.prototype, {
+    load: function (url, callback, asset) {
+        var options = {
+            cache: true,
+            responseType: "arraybuffer",
+            retry: this.retryRequests
+        };
+        pc.http.get(url.load, options, callback);
+    },
+
+    open: function (url, data, device) {
+        var textureData = this.parse(data);
+
+        if (!textureData) {
+            return null;
+        }
+
+        var texture = new pc.Texture(device, {
+            name: url,
+            // #ifdef PROFILER
```

Comment:
Presumably, the preprocessor statements will just get ignored here.

Reply:
yeah, as you can see this was extracted from the engine. I'd prefer to leave the preprocessor stuff for now unless you think it's objectionable...

Reply:
Well, it's totally harmless - they'll just be treated as comments. But might be a bit confusing for developers who aren't familiar with what it's for.